### PR TITLE
Drop support for EOL Python 3.3 and 3.4, add Python 3.7

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/zopefoundation/z3c.coverage/issues/10.

Python 3.3 is EOL and no longer receiving security updates (or any updates) from the core Python team.

Version | Release date | Supported until
--- | ---------- | ----------
3.3 | 2012-09-29 | 2017-09-29

Source: https://en.wikipedia.org/wiki/CPython#Version_history

It's also little used, and 3.3 is failing the build.

This also Upgrades Python syntax with [pyupgrade](https://github.com/asottile/pyupgrade), and add support for Python 3.7, which needs a little workaround on Travis CI: https://github.com/travis-ci/travis-ci/issues/9815.

### CODE REVIEW

**Comment by hugovk** [comment]
> Can you please add a CHANGES.rst entry saying "Dropped Python 3.3 support."?
> 
> Other than that, LGTM.

@mgedmin Hi, thanks for the review! Sorry, only just saw your comment now, and I've updated it!

**Comment by mgedmin** [comment]
BTW have you signed the [Zope Foundation Committer Agreement](https://docs.zope.org/developer/becoming-a-committer.html)?  I'm not allowed to merge code from people who haven't signed it.  (And once you sign it, you'll get the committer bit that allows you to hit the nice big green Merge button yourself.)

**Comment by hugovk** [comment]
I've not signed yet, is it possible to do it electronically? If not, I can do it on Friday when I next have access to a printer :)

**Comment by jugmac00** [comment]
Hm, the above mentioned pdf seems to be outdated. 

This is the updated one
https://github.com/zopefoundation/developer_docs/blob/master/source/Zope_Foundation_Committer_Agreement-20181008.pdf

**Comment by hugovk** [comment]
Thanks, who may I put as the reference committer?

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,12 @@ python:
     - 3.6
     - pypy
     - pypy3
+# Enable 3.7 without globally enabling sudo and dist: xenial for other build jobs
+matrix:
+  include:
+    - python: 3.7
+      dist: xenial
+      sudo: true
```
Comment (jugmac00) [inline .travis.yml]: The usage of sudo is deprecated, cf
https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

**Comment by jugmac00** [comment]
> Thanks, who may I put as the reference committer?

I am only a commiter like since half a year and I do not think @tseaver or @jimfulton know me.

@mgedmin Can you jump in here?

File: .travis.yml (hunk 1 above)
Comment (hugovk) [reply .travis.yml]: Removed!

**Comment by mgedmin** [comment]
@hugovk, feel free to put me as a reference committer.

**Comment by hugovk** [comment]
I've just emailed the agreement, sorry for the delay. I've also updated this PR to drop Python 3.4, which went EOL yesterday (2019-03-16); let me know if you need to keep it longer. https://en.wikipedia.org/wiki/CPython#Version_history

**Comment by icemac** [comment]
Parts of this PR were fixed by #12.
@hugovk Did your committer agreement get accepted?

**Comment by hugovk** [comment]
> @hugovk Did your committer agreement get accepted?

I've no idea, I never got a reply. There's nothing in my spam.

**Comment by hugovk** [comment]
There's not much left of this PR now #12 is merged! I'm happy to just drop this PR now if that's easier.

**Comment by jugmac00** [comment]
Hi @hugovk, sorry for forgetting about your pull request - I saw the broken build on TravisCI and automatically created a pull request to fix it.

As of the committer agreement - @icemac has good connections to the admins. Maybe he can ask what happened with your application.

**Comment by hugovk** [comment]
@jugmac00 No problem, it's important to get the build back to green quickly :)

**Comment by icemac** [comment]
The remaining changes are so minimal that I am going to merge them right anyway.

**Pull request merged**